go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -149,6 +156,73 @@ func delPidfile(pidfile string) {
 	}
 }
 
+// ovnkubeControllerSyncPostBootFilePath is located in storage that does not persist following a reboot. ovn-controller start
+// is predicated on the existence of this file.
+const ovnkubeControllerSyncPostBootFilePath = "/tmp/ovnkube-controller-sync-post-boot"
+
+func ensureOVNKubeControllerSyncPostBootFile() error {
+	if err := os.WriteFile(ovnkubeControllerSyncPostBootFilePath, []byte(time.Now().String()), 0o644); err != nil {
+		return fmt.Errorf("failed to write to file at path %q: %v", ovnkubeControllerSyncPostBootFilePath, err)
+	}
+	return nil
+}
+
+// doesOVNKubeControllerSyncPostBootFileExist checks if a sentinel file exists. File will not exist post Node restart. OVNKube controller creates the file post sync and after a reboot.
+func doesOVNKubeControllerSyncPostBootFileExist() (bool, error) {
+	exists := false
+	_, err := os.Stat(ovnkubeControllerSyncPostBootFilePath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return exists, fmt.Errorf("failed to stat file at path %q: %w", ovnkubeControllerSyncPostBootFilePath, err)
+		}
+		exists = true
```

Comment:
this is bit confusing, exists flag is set to true when file doesn't exist, but the method name suggests otherwise.